### ISSUE TITLE
Position doesn't always respond to changes in the ViewModel

### ISSUE THREAD

**Issue by John-Burbidge** [xamarin/Xamarin.Forms.CarouselView#8]
I'm using two way data binding on the position property.   It mostly works but intermittently it will not respond to the view ViewModel changes (even though I can see that the ViewModel has changed).

**Comment by JaridKG** [comment]
@John-Burbidge have you created a bug-zilla for this? I am seeing this as well

**Comment by Dids** [comment]
I thought this was just me. Also seeing duplicate views (ie. identical content when swiping across views).

**Comment by Dids** [comment]
From what I can tell, it stops binding binding to the Position property, at least that's what it seems like.
It's most likely just not registering the view changes properly, and then delegating the Position property changes through the binding.
Still, makes working with it a nightmare.

EDIT: In my case, my bound source changes quite drastically from time to time, so I usually set Position to 0 when it does. It doesn't seem to always redraw the view when the source changes though.

**Comment by JaridKG** [comment]
I am not seeing duplicate views, however I am seeing the position problem @Dids. I do a backwards navigation to 0 in the onAppearing of the carousel (say you navigate forwards and back to a page that has it) and it works 99% of the time. But every once in a while indicators I have built that are also bound to the position don't react accordingly

**Comment by Dids** [comment]
I feel like this is related to changing the Position, which triggers the carousel animation, then changing the Position again before the animation has finished, which leaves it in an inconsistent state.

**Comment by EmmanVazz** [comment]
I also see the binding issue with the position not getting updated properly.

**Comment by puna** [comment]
@Dids regarding your "duplicate views", is it possible you dynamically insert items before the currently active item on iOS 10? And during this "duplication" an item is also skipped?
Bcs this is what is happening to me in:  #22 .

**Comment by Dids** [comment]
@puna My issues are mostly gone now.

Essentially, my issue was that the CarouselView is clearly not designed for a scenario where both the items and index changes a lot, so my solution was to combine all my different data sets and bind it to CarouselView once. No more duplicates or crashes.

**Comment by hardcodet** [comment]
Same here - the case is super simple:

ViewModel exposes 
*  List[Foo], which is the carousel's ItemsSource
* SelectedFooIndex property, which has Two-Way binding from the carousel's Position

I'm resetting the List initially, which causes the carousel to render properly. Subsequent programmatic changes to the SelectedFooIndex property are not reliably registered by the Carousel though. The binding is still intact though - when swiping the carousel, this updates the view model. (And yes, I'm firing the right property change event in my property's setter ;)

**Comment by Dids** [comment]
@hardcodet I'm now seeing the exact same thing. Right now, I'm basically raising INotifyPropertyChanged manually after the carousel view has appeared, which is not a solution from a user's point of view, as it shows the old (always first index) for a second or two, then jumps to the correct one. After doing this, scrolling breaks when programmatically changing the index, and I have to swipe scroll by hand to fix it. Firing the property change notification when the carousel isn't visible has no impact visually.

I'm fairly certain that this issue is the one and the same for all of us. The binding works, but isn't being applied when the view isn't visible.

**Comment by Dids** [comment]
I wonder if this PR fixes it:
https://github.com/xamarin/Xamarin.Forms.CarouselView/pull/9

**Comment by nithinshiriya** [comment]
@Dids  Do you guys have any solution for this.

I am using PRSIM view model. I want to set the option from the view model. But it doesn't work. It always takes default position to zero.

**Comment by Dids** [comment]
@nithinshiriya Nope, unless you count setting the data after the view has loaded. That works, but produces a visual delay and breaks some things.

**Comment by John-Burbidge** [comment]
@nithinshiriya I'm using Prism and it's working fine

**Comment by John-Burbidge** [comment]
I was able to fix the issue where changing the Position property did not always cause the CarouselView to call ScrollToPosition.

I'll make a PR on this to illustrate what I did.

**Comment by Dids** [comment]
@John-Burbidge Last I tried building this project, it didn't quite work with VS2017, so I'm assuming you're on VS2015?

Would be awesome if a fix was found and someone pushed a forked nuget package. A proper maintained fork would be even better though. As it stands, the current version is causing nothing but headache for me, on both iOS and Android.

**Comment by John-Burbidge** [comment]
https://github.com/John-Burbidge/Xamarin.Forms.CarouselView

**Comment by John-Burbidge** [comment]
My efforts were only focused on iOS as we're only going iOS on this project.   Conceivably Android has a similar problem if the same symptom of CarouselViews not always Scrolling to the view of the changed position.

I think that in the next quarter the CarouselView will be updated with a greatly improved stable release version.

So really this is just a stop gap until then.

Another thing that I removed (not included here because it's a feature not a bug) was the animation between intermediate views.

Say switching from position 0 to position  12, animating the intermediate views was slowing the application in my case so I disabled the animation.   Swipe gestures still animate.
